windows/src/desktop/setup/Keyman.Setup.System.OnlineResourceCheck.pas

```diff
@@ -0,0 +1,103 @@
+{**
+ * Wrapper for checking api.keyman.com for latest version of resources to be
+ * installed. Uses the /windows/14.0/update endpoints.
+ *}
+unit Keyman.Setup.System.OnlineResourceCheck;
+
+interface
+
+uses
+  Keyman.Setup.System.InstallInfo;
+
+type
+  TOnlineResourceCheck = class sealed
+    class procedure QueryServer(ASilent: Boolean; AInstallInfo: TInstallInfo); static;
+  end;
+
+implementation
+
+uses
+  System.Net.HttpClient,
+  System.Net.URLClient,
+  System.SysUtils,
+
+  GlobalProxySettings,
+  Keyman.System.UpdateCheckResponse,
+  KeymanVersion,
+  PackageInfo,
+  Upload_Settings,
+  versioninfo;
+
+class procedure TOnlineResourceCheck.QueryServer(ASilent: Boolean; AInstallInfo: TInstallInfo);
+var
+  http: THTTPClient;
+  pack: TInstallInfoPackage;
+  ucr: TUpdateCheckResponse;
+  ucrpack: TUpdateCheckResponsePackage;
+  location: TInstallInfoFileLocation;
+  packLocation: TInstallInfoPackageFileLocation;
+  currentVersion: string;
+  lang: TUpdateCheckResponseLanguage;
+  iipl: TInstallInfoPackageLanguage;
+  url: TURI;
+  response: IHTTPResponse;
+  u: AnsiString;
+begin
+  currentVersion := AInstallInfo.MsiLocations.LatestVersion(SKeymanVersion_Min_Evergreen);
+
+  url := TURI.Create(MakeAPIURL(API_Path_UpdateCheck_Windows));
+  url.AddParameter('version', currentVersion);
+  url.AddParameter('tier', KeymanVersion.CKeymanVersionInfo.Tier);
+  for pack in AInstallInfo.Packages do
+    url.AddParameter('package_'+pack.ID, pack.Locations.LatestVersion);
+  http := THTTPClient.Create;
+  try
+    response := http.Get(url.ToString);
+    if response.StatusCode <> 200 then
+    begin
+      // TODO: log failed response
+      //raise EOnlineUpdateCheck.Create('Error '+IntToStr(Response.StatusCode));
+      Exit;
+    end;
```

Comment:
I assume you're noting this for implementation in the near future in some manner?

Reply:
Yes, all TODO items are noted by me (didn't want to create issues for each one of them -- it's just too much noise).